### PR TITLE
aloe medical plant addition

### PR DESCRIPTION
**Pull request by Tlaltecuhtli**
:cl:
add: aloe plant
add: aloe plant cream (microwaved /bonfired aloe)
add: advanced healing mesh (made with aloe juice + cellulose + sterilizine)
add: lowered price of polypypyr suture to require  less formaldeyhde and less popolypyyr
/:cl:

new medicine that botany can make that they can supply med with that isnt just wheat with chem x y z 

aloe cream works like the suture things it heals 3,3 instead of 10

advanced mesh is the equivalent of the polypryr suture thing for burns

if you have interesting changes ideas write below

### CODE REVIEW

**Comment by spookydonut** [comment]
Might help if you made sure it compiles before committing/opening a pr

**Comment by Tlaltecuhtli** [comment]
might help if prs dont get merged after i made a branch but i dont go comment on oher people's prs about it

File: code/game/objects/items/stacks/medical.dm [hunk 1]
```diff
@@ -285,6 +284,55 @@
...
+	icon_state = "aloe_paste"
+	self_delay = 20
+	other_delay = 10
+	novariants = TRUE
+	amount = 20
+	max_amount = 20
+	var/heal = 3
+	grind_results = list(/datum/reagent/consumable/aloejuice = 1)
+
+/obj/item/stack/medical/aloe/heal(mob/living/M, mob/user)
```
Comment (cacogen) [inline code/game/objects/items/stacks/medical.dm]: should you have to do this?

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm [hunk 2]
```diff
@@ -881,3 +881,18 @@
 	glass_icon_state = "glass_yellow"
 	glass_name = "glass of bungo juice"
 	glass_desc = "Exotic! You feel like you are on vactation already."
+
+/datum/reagent/consumable/aloejuice
+	name = "Aloe Juice"
+	color = "#A3C48B"
+	description = "An healthy and refreshing juice."
```
Comment (cacogen) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm]:
```suggestion
	description = "A healthy and refreshing juice."
```

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm [hunk 3]
```diff
@@ -881,3 +881,18 @@
...
+	taste_description = "vegetable"
+	glass_icon_state = "glass_yellow"
+	glass_name = "glass of aloe juice"
+	glass_desc = "An healthy and refreshing juice."
```
Comment (cacogen) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm]:
```suggestion
	glass_desc = "A healthy and refreshing juice."
```

File: code/game/objects/items/stacks/medical.dm [hunk 4]
```diff
@@ -285,6 +284,55 @@
...
+
+/obj/item/stack/medical/mesh/advanced/update_icon_state()
+	if(!is_open)
+		icon_state = "aloe_mesh_closed"
+	else
+		return ..()
+
+/obj/item/stack/medical/aloe
+	name = "aloe cream"
+	desc = "An healing paste you can apply on wounds."
```
Comment (cacogen) [inline code/game/objects/items/stacks/medical.dm]:
```suggestion
	desc = "A healing paste you can apply on wounds."

```

File: code/game/objects/items/stacks/medical.dm [hunk 5]
```diff
@@ -285,6 +284,55 @@
 		return
 	. = ..()
 
+/obj/item/stack/medical/mesh/advanced
+	name = "advanced regenerative mesh"
+	desc = "An advanced mesh made with aloe extracts and sterilizing chemicals, used to threat burns."
```
Comment (cacogen) [inline code/game/objects/items/stacks/medical.dm]:
```suggestion
	desc = "An advanced mesh made with aloe extracts and sterilizing chemicals, used to treat burns."

```

**Comment by FantasticFwoosh** [comment]
Sprites are not great, i have some better ones for you to use from other projects i have had a hand in, so shoot me a DM some time if  you want them. Eitherway nice work.

agave agave
(etc. full set of growth sprites too, can improv a seed packet if asked)

File: code/game/objects/items/stacks/medical.dm (hunk 1 above)
Comment (Tlaltecuhtli) [reply code/game/objects/items/stacks/medical.dm]: the parent doesnt have heals and a ssubtype of subtype is gonna rewrite it anyway

**Comment by trollbreeder** [comment]
Alternative medicine is finally catching up to Big Pharma.

**Comment by Tlaltecuhtli** [comment]
are we talking about the same plant a

**Comment by ATH1909** [comment]
Can we finally make sutures easier to make from polypyr? They currently cost an absurd amount of chems per suture, IIRC.

**Comment by Krysonism** [comment]
> 
> 
> Can we finally make sutures easier to make from polypyr? They currently cost an absurd amount of chems per suture, IIRC.

I do support this, i thought Gaia would be hit harder and i would be able to implement more uses for the plant.

File: code/game/objects/items/stacks/medical.dm [hunk 6]
```diff
@@ -285,6 +284,57 @@
...
 
+/obj/item/stack/medical/mesh/advanced
+	name = "advanced regenerative mesh"
+	desc = "An advanced mesh made with aloe extracts and sterilizing chemicals, used to treat burns."
+
+	gender = PLURAL
+	singular_name = "advanced regenerative mesh"
+	icon_state = "aloe_mesh"
+	heal_burn = 15
+	grind_results = list(/datum/reagent/medicine/spaceacillin = 1)
```
Comment (nemvar) [inline code/game/objects/items/stacks/medical.dm]: Grind result doesn't really make sense, considering what is required to mix this.

**Comment by ExcessiveUseOfCobblestone** [comment]
mc

File: code/game/objects/items/stacks/medical.dm [hunk 7]
```diff
@@ -285,6 +284,57 @@
...
+	amount = 20
+	max_amount = 20
+	var/heal = 3
+	grind_results = list(/datum/reagent/consumable/aloejuice = 1)
+
+/obj/item/stack/medical/aloe/heal(mob/living/M, mob/user)
+	. = ..()
+	if(M.stat == DEAD)
+		to_chat(user, "<span class='warning'>[M] is dead! You can not help [M.p_them()].</span>")
+		return
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/items/stacks/medical.dm]:
```suggestion
		return FALSE
```

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm [hunk 8]
```diff
@@ -899,3 +899,19 @@
...
+	taste_description = "vegetable"
+	glass_icon_state = "glass_yellow"
+	glass_name = "glass of aloe juice"
+	glass_desc = "A healthy and refreshing juice."
+
+/datum/reagent/consumable/aloejuice/on_mob_life(mob/living/M)
+	if(M.getToxLoss() && prob(30))
+		M.adjustToxLoss(-1, 0)
+	..()
+	. = 1
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm]:
```suggestion
	. = TRUE
```

File: code/modules/reagents/chemistry/recipes/medicine.dm [hunk 9]
```diff
@@ -155,10 +155,19 @@
...
 	return
+
+/datum/chemical_reaction/medmesh
+	required_reagents = list(/datum/reagent/cellulose = 20, /datum/reagent/consumable/aloejuice = 20, /datum/reagent/space_cleaner/sterilizine = 10)
+
+/datum/chemical_reaction/medmesh/on_reaction(datum/reagents/holder, created_volume)
+	var/location = get_turf(holder.my_atom)
+	for(var/i = 1, i <= created_volume, i++)
+		new /obj/item/stack/medical/mesh/advanced(location)
+	return
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/recipes/medicine.dm]:
```suggestion
```

File: code/modules/reagents/chemistry/recipes/medicine.dm [hunk 10]
```diff
@@ -155,10 +155,19 @@
...
 
 /datum/chemical_reaction/medsuture
-	required_reagents = list(/datum/reagent/cellulose = 10, /datum/reagent/toxin/formaldehyde = 30, /datum/reagent/medicine/polypyr = 30) //This might be a bit much, reagent cost should be reviewed after implementation.
+	required_reagents = list(/datum/reagent/cellulose = 10, /datum/reagent/toxin/formaldehyde = 20, /datum/reagent/medicine/polypyr = 15) //This might be a bit much, reagent cost should be reviewed after implementation.
 
 /datum/chemical_reaction/medsuture/on_reaction(datum/reagents/holder, created_volume)
 	var/location = get_turf(holder.my_atom)
 	for(var/i = 1, i <= created_volume, i++)
 		new /obj/item/stack/medical/suture/medicated(location)
 	return
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/recipes/medicine.dm]:
```suggestion
```

**Comment by ExcessiveUseOfCobblestone** [comment]
haha wh00psie :D

**Comment by ExcessiveUseOfCobblestone** [comment]
I can rerun travis you don't have to make dummy commits

File: code/modules/hydroponics/grown/misc.dm [hunk 11]
```diff
@@ -228,3 +228,34 @@
...
+	plantname = "Aloe"
+	product = /obj/item/reagent_containers/food/snacks/grown/aloe
+	lifespan = 60
+	endurance = 25
+	maturation = 4
+	production = 4
+	yield = 6
+	growthstages = 5
+	growing_icon = 'icons/obj/hydroponics/growing_vegetables.dmi'
+	reagents_add = list(/datum/reagent/consumable/nutriment/vitamin = 0.06, /datum/reagent/consumable/nutriment = 0.05)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/hydroponics/grown/misc.dm]:
```suggestion
	reagents_add = list(/datum/reagent/consumable/nutriment/vitamin = 0.05, /datum/reagent/consumable/nutriment = 0.05)
```

**Pull request merged**